tests/foreman/cli/test_contentview.py

```diff
@@ -2069,6 +2077,159 @@ def test_positive_subscribe_chost_by_id_using_puppet_content(self):
+                permission['name']
+                for permission in available_rc_permissions[resource_type]
+                if permission['name'] in permission_names
+            ]
+            # assert that all the required permissions are available
+            self.assertEqual(set(permission_names),
+                             set(available_permission_names))
+            # Create the current resource type role permissions
+            make_filter({
+                'role-id': role['id'],
+                'permissions': permission_names,
+            })
+        # Add the created and initiated role with permissions to user
+        User.add_role({'id': user['id'], 'role-id': role['id']})
+        # assert that the user is not an admin one and cannot read the current
+        # role info (note: view_roles is not in the required permissions)
+        with self.assertRaises(CLIReturnCodeError) as context:
+            Role.with_user(user_name, user_password).info(
+                {'id': role['id']})
+        self.assertIn(
+            'Forbidden - server refused to process the request',
+            context.exception.stderr
+        )
+        # get the default lifecycle environment info
+        env = LifecycleEnvironment.info(
+            {'organization-id': org['id'], 'name': ENVIRONMENT})
+        # Create a product
+        product = make_product({'organization-id': org['id']})
+        # Create a yum repository and synchronize
+        repo = make_repository({
+            'product-id': product['id'],
+            'url': FAKE_1_YUM_REPO
+        })
+        Repository.synchronize({'id': repo['id']})
+        # Create a content view, add the yum repository and publish
+        content_view = make_content_view({u'organization-id': org['id']})
+        ContentView.add_repository({
+            'id': content_view['id'],
+            'organization-id': org['id'],
+            'repository-id': repo['id'],
+        })
+        ContentView.publish({u'id': content_view['id']})
+        content_view = ContentView.info({u'id': content_view['id']})
+        # assert that the content view has been published and has versions
+        self.assertGreater(len(content_view['versions']), 0)
+        # assert that the user can read the content view info as per required
+        # permissions
+        user_content_view = ContentView.with_user(
+            user_name, user_password).info({'id': content_view['id']})
+        # assert that this is the same content view
+        self.assertEqual(content_view['name'], user_content_view['name'])
+        # create a client host and register it with the created user
+        with VirtualMachine(distro=DISTRO_RHEL7) as host_client:
+            host_client.install_katello_ca()
+            cmd = (
+                u'subscription-manager register --org {0} --environment {1}'
+                u' --username {2} --password {3} --force'
+                .format(org['name'], env['name'], user_name, user_password)
```

Comment:
How about updating `register_contenthost()` with optional username and password instead? Changes to the body of method will be very trivial like `s/settings.server.admin_username/username or settings.server.admin_username/` but in the end you'll use common helper and basically we can reuse it later in other tests instead of passing raw command each time.